File: SavvyApp/Packages/Core/SavvyFoundation/Sources/SavvyFoundation/Models/Savings.swift
import Foundation

public struct Savings: FirestoreEntity {
    public let id: String
    public var amount: Decimal
    public var category: SavingsCategory
    public var goalId: String?
    public var note: String?
    public var date: Date
    public var status: SavingsStatus
    public var isDeleted: Bool
    public let createdAt: Date

    public init(
        id: String = UUID().uuidString,
        amount: Decimal,
        category: SavingsCategory,
        goalId: String? = nil,
        note: String? = nil,
        date: Date = Date(),
        status: SavingsStatus = .active,
        isDeleted: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.amount = amount
        self.category = category
        self.goalId = goalId
        self.note = note
        self.date = date
        self.status = status
        self.isDeleted = isDeleted
        self.createdAt = createdAt
    }
}

File: lib/features/savings/domain/models/savings.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:savvy/core/constants/financial_enums.dart';

part 'savings.freezed.dart';
part 'savings.g.dart';

@freezed
abstract class Savings with _$Savings {
  const factory Savings({
    required String id,
    required double amount,
    @Default(SavingsCategory.other) SavingsCategory category,
    String? title,
    String? goalId,
    String? note,
    required DateTime date,
    @Default(SavingsStatus.active) SavingsStatus status,
    @Default(false) bool isDeleted,
    required DateTime createdAt,
  }) = _Savings;

  factory Savings.fromJson(Map<String, dynamic> json) =>
      _$SavingsFromJson(json);
}
